ext/dom/html5_serializer.c

```diff
@@ -70,7 +70,20 @@ static zend_result dom_html5_escape_string(dom_html5_serialize_context *ctx, con
 {
 	const char *last_output = content;
 
-	while (*content != '\0') {
+	while (true) {
+		size_t chunk_length;
+		/* Note: uses UTF-8 internally, so <C2 A0> indicates a non-breaking space */
+		if (attribute_mode) {
+			chunk_length = strcspn(content, "&\xC2\"");
+		} else {
+			chunk_length = strcspn(content, "&\xC2<>");
+		}
```

Comment:
nits

I thought that defining the mask outside the loop would reduce the number of if operations. What do you think?

Reply:
Makes sense to me, thx for checking.